File: library/src/main/java/io/vithor/yamvpframework/mvp/presenter/GraphPresenter.java
package io.vithor.yamvpframework.mvp.presenter;

import android.support.annotation.Nullable;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Hazer on 1/10/16.
 */
public abstract class GraphPresenter {
    private List<GraphPresenter> childPresenters = new ArrayList<>(4);
    private @Nullable GraphPresenter parent;

    public GraphPresenter() {
    }

    public GraphPresenter(@Nullable GraphPresenter parent) {
        this.parent = parent;
    }

    public void addChildPresenter(GraphPresenter presenter) {
        if (!childPresenters.contains(presenter)) {
            childPresenters.add(presenter);
        }
    }

    public void removeChildPresenter(GraphPresenter presenter) {
        childPresenters.remove(presenter);
    }

    public void notifyChildPresenters() {
        for (GraphPresenter presenter: childPresenters) {

        }
    }
}

File: mvp/src/main/java/io/vithor/yamvpframework/mvp/presenter/GraphPresenter.kt
package io.vithor.yamvpframework.mvp.presenter

import java.util.ArrayList

/**
 * Created by Vithorio Polten on 1/10/16.
 */
abstract class GraphPresenter(parent: GraphPresenter?) {
    private val childPresenters = ArrayList<GraphPresenter>(4)
    private val parent: GraphPresenter? = null

    fun addChildPresenter(presenter: GraphPresenter) {
        if (!childPresenters.contains(presenter)) {
            childPresenters.add(presenter)
        }
    }

    fun removeChildPresenter(presenter: GraphPresenter) {
        childPresenters.remove(presenter)
    }

    fun notifyChildPresenters() {
        for (presenter in childPresenters) {

        }
    }
}
